### PR TITLE
fix(heading-order): evaluate headings from iframes in DOM order

### PR DESCRIPTION
**Pull request by straker**
**NOTE:** this changes the selectors returned by axe as iframe results are now in DOM order so the first node encountered could be an iframe element rather than the top-level element.

Closes issue: #2561

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**

- [x] Follows the commit message policy, appropriate for next version
- [x] Code is reviewed for security

### CODE REVIEW

File: lib/core/utils/merge-results.js [hunk 1]
```diff
@@ -104,6 +104,28 @@ function mergeResults(frameResults, options) {
 			}
 		});
 	});
+
+	// don't sort results if we don't have the ability to run
```
Comment (WilcoFiers) [inline lib/core/utils/merge-results.js:108]:
```suggestion
	// Only sort results if we have the ability to run
```

**Review by WilcoFiers — APPROVED** [review]

File: test/integration/rules/heading-order/heading-order.json [hunk 2]
```diff
@@ -8,6 +8,11 @@
 		["#heading3"],
 		["#heading4"],
 		["#heading5"],
-		["#heading7"]
+		["#heading7"],
+		["#heading8"],
+		["#heading9"],
+		["#heading10"],
```
Comment (dylanb) [inline test/integration/rules/heading-order/heading-order.json:14]: `#heading10` should come after the frame

File: test/integration/rules/heading-order/heading-order.json [hunk 3]
```diff
@@ -8,6 +8,11 @@
...
+		["iframe", "#frame-heading1"],
+		["iframe", "#frame-heading2"]
```
Comment (WilcoFiers) [inline test/integration/rules/heading-order/heading-order.json:16]:
```suggestion
		["iframe", "#frame-heading1"],
		["iframe", "#frame-heading2"],
		["#heading10"]
```

File: test/integration/rules/heading-order/heading-order.json (hunk 2 above)
Comment (WilcoFiers) [reply test/integration/rules/heading-order/heading-order.json:14]: The test results aren't ordered. Nevertheless, probably good to put them in order.

**Review by WilcoFiers — DISMISSED** [review]

**Review by WilcoFiers — APPROVED** [review]

**Pull request merged**